.github/workflows/image-update.yaml

```diff
@@ -10,15 +10,50 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
       - uses: actions/checkout@v1
-      - run: make -f Makefile.docker build-image
+      # Login to both registries
       - run: echo ${DOCKER_HUB_PASSWORD} | docker login --username weaveworkseksctlci --password-stdin
         env:
           DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
-      - run: make -f Makefile.docker push-build-image
+      - name: Update the build image, push it Docker Hub and update manifest in the repo
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+        run: |
+          set -x
+
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          # Update image manifest and push new image to registry only when needed
+          if ! make -f Makefile.docker check-build-image-manifest-up-to-date ; then
```

Comment:
If we move this `if ! make -f Makefile.docker check-build-image-manifest-up-to-date...` to its own `run:` step maybe we achieve this if and we can use GHA push instead of the curl command below?

something like:

```
- name: Check image is up to date
  run: ! make -f Makefile.docker check-build-image-manifest-up-to-date
- name: Update manifest and push image to Docker Hub 
  run: |
    set -x
    ...
    # Commit changes to the manifest
    git config --local user.email "weaveworksbot@users.noreply.github.com"
    config --local user.name "Weaveworks Bot"
    .....
```
Then we can use the pre-defined action that enables you to open a PR?

That is assuming that when a `run` step fails it will not continue with the other steps.

Reply:
Ah, I see what you mean. Good idea, but one issue is that when `make -f Makefile.docker check-build-image-manifest-up-to-date` returns 0 (which is going to happen most often), the job will be marked as failing.

Reply:
> Then we can use the pre-defined action that enables you to open a PR?

Ideally, yes, but one that I found took would imply a few lines of config. I can have another look into this, but `curl` option didn't seem too problematic. The only issue to me is that the script is a bit verbose, and the error handling is subject to the usual concerns with bash...